model/model.js

```diff
@@ -25,15 +25,37 @@ readUsers = () => {
     });
 };
 
-readReviews = () => {
-  let queryString = `SELECT reviews.review_id, reviews.owner, reviews.title, reviews.category, review_img_url, reviews.created_at, reviews.votes, reviews.designer, (SELECT COUNT(*) FROM comments WHERE comments.review_id = reviews.review_id) AS comment_count
+readReviews = (sort_by = "created_at", order = "desc", category) => {
+  let queryString = `SELECT reviews.*, CAST(COUNT(comments.review_id) AS int) AS comment_count
   FROM reviews
-  JOIN comments ON reviews.review_id = comments.review_id
-  GROUP BY reviews.review_id
-  ORDER BY created_at DESC
-  ;`;
-  return db.query(queryString).then((results) => {
-    return results.rows;
+  JOIN comments ON reviews.review_id = comments.review_id `;
+  const queryArray = [];
+  const columnArray = [
+    "owner",
+    "title",
+    "designer",
+    "review_img_url",
+    "review_body",
+    "category",
+    "created_at",
+    "votes",
+  ];
+  const orderArray = ["asc", "desc"];
+  if (category) {
+    queryString += `WHERE category = $1`;
+    queryArray.push(category);
+  }
+  if (!columnArray.includes(sort_by) || !orderArray.includes(order)) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+
+  queryString += ` GROUP BY reviews.review_id
+  ORDER BY ${sort_by} ${order};`;
+  return db.query(queryString, queryArray).then((res) => {
+    if (res.rows.length === 0) {
+      return Promise.reject({ status: 404, msg: "Category not found" });
+    }
```

Comment:
I think this part of your code will need to change when you update your tests.

Reply:
That makes sense will sort all changes! thanks for your help!

Reply:
Im struggling with this, I have this working now, but it also returns a 200 for anything random entered into the query category=dsjlflf returns an empty array rather than an error any ideas?

Reply:
Sorted it :)